### PR TITLE
Shuffle the vertex pair 

### LINKED ISSUE #3001: [BUG]: vertex pair not properly shuffled
### Version

22.12

### Which installation method(s) does this occur on?

Docker, Conda, Pip, Source

### Describe the bug.

The vertex pairs are not properly shuffled to the appropriate GPUs leading to undefined behaviors such as illegal memory accesses. This causes all the MG similarity algorithms (`jaccard`, `sorensen`, `overlap`) to fail at 8+ GPUs

### Minimum reproducible example

```shell
Run the script below or any MG similarity algos tests with 8+ GPUs

setup_objs = setup()
    client = setup_objs[0]
    num_workers = len(client.scheduler_info()['workers'])

    df = karate.get_edgelist()
    
    ddf = dask_cudf.from_cudf(df, npartitions=num_workers)

    # Create MG Graph
    dg = cugraph.Graph(directed=False)
    dg.from_dask_cudf_edgelist(
        ddf, source='src', destination='dst',
        legacy_renum_only=True)

    # Get vertex_pair by computing the two_hop_neighbors
    vertex_pair = dg.get_two_hop_neighbors()
    vertex_pair = vertex_pair.compute.head()

    # Call jaccard
    df = dcg.jaccard(dg, vertex_pair)    

    teardown(*setup_objs)
```
```


### Relevant log output

```shell
Exception: "RuntimeError('non-success value returned from cugraph_jaccard_coefficients: CUGRAPH_UNKNOWN_ERROR std::bad_alloc: out_of_memory: CUDA error at: /gpfs/fs1/projects/sw_rapids/users/jnke/miniconda3/envs/cugraph_test/include/rmm/mr/device/cuda_memory_resource.hpp')"
```
```


### Environment details

_No response_

### Other/Misc.

_No response_

### Code of Conduct

- [X] I agree to follow cuGraph's Code of Conduct
- [X] I have searched the [open bugs](https://github.com/rapidsai/cugraph/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

### PR DESCRIPTION
**Pull request by jnke2016**
An illegal memory access occurs when running the MG similarity algos at certain scale. This is caused by vertex pairs not being shuffled appropriately.
This PR:
1. Shuffle the vertex pairs based on the edge partitioning
2. Update the the vertex pairs column names which are not necessarily edgelists
3. Update the docstrings, tests and notebooks accordingly

closes #3001

### CODE REVIEW

**Comment by jnke2016** [comment]
rerun tests

**Review by rlratzel — APPROVED** [review]
LGTM

File: notebooks/algorithms/link_prediction/Jaccard-Similarity.ipynb [hunk 1]
```diff
@@ -59,9 +59,9 @@
     "Returns:\n",
     "\n",
     "    df: cudf.DataFrame with three names columns:\n",
-    "        df[\"source\"]: The source vertex id.\n",
-    "        df[\"destination\"]: The destination vertex id.\n",
-    "        df[\"jaccard_coeff\"]: The jaccard coefficient computed between the source and destination vertex.\n",
+    "        df[\"first\"]: The first vertex id of each pair.\n",
+    "        df[\"second\"]: The second vertex i of each pair.\n",
```
Comment (ChuckHastings) [inline notebooks/algorithms/link_prediction/Jaccard-Similarity.ipynb:63]: Typo:  "i" should be "id"

**Review by ChuckHastings — APPROVED** [review]
One minor documentation typo.

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**